core/src/main/scala/cats/eq/Is.scala

```diff
@@ -0,0 +1,64 @@
+package cats.eq
+
+import cats.Id
+
+/**
+ * A value of `A Is B` witnesses that the types `A` and `B` are the same.
+ * You can use this value to coerce `A` into `B` and therefore `A Is B` is
+ * similar in feel to `A =:= B`; however, `A Is B` has more computational
```

Comment:
It's _true_ that `A =:= B` ought to be enough to say that `F[A] =:= F[B]` but there's no internal mechanism to furnish proof of that. We have to believe this assertion by using knowledge and process _about_ Scala instead of _within_ Scala. This makes it harder to do type shenanigans which require notions of equality unless you're willing to do a lot of  hand-validated `asInstanceOf` calls.

So, the other way of saying this is to say that `A =:= B` produces two functions `f: A => B` and `g: B => A` such that `x = f(g(x))` for all `x`. This is not enough to get `F[A] =:= F[B]` directly even just for some `F`s since there's no way to use `f` and `g` to produce the required witnesses for just any `F` we can dream up. Immediately, `F` might be abstract and then we're done.

On the other hand, `A Is B` can trivially prove `A =:= B` (as shown) because having `to[F](f: F[A]): F[B]` and `fro[F](f: F[B]): F[A]` easily recovers `f = to[Id]` and `g = fro[Id]`. Generally this power is magnified with type lambdas (and heavily simplified with kind projector) so let us do what we sort of empirically want: use equalities to substitute out type variables in arbitrary type expressions.

So it's reasonably crucial to use `Is` to work with type equalities unless one wants to fall to coercions (which may be erroneous).